File: EyeGuideAndroid/app/src/main/java/com/ZacharyZhang/eyeguide/data/model/UsageStat.kt
package com.ZacharyZhang.eyeguide.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class UsageStat(
    val id: String,
    @SerialName("user_id") val userId: String,
    val feature: String,
    val timestamp: String,
    val duration: Int? = null,
    val success: Boolean,
    @SerialName("error_message") val errorMessage: String? = null,
)

@Serializable
data class RecordUsageRequest(
    val feature: String,
    val duration: Int? = null,
    val success: Boolean,
    @SerialName("error_message") val errorMessage: String? = null,
)

@Serializable
data class UsageStatResponse(
    val stat: UsageStat,
)

@Serializable
data class UsageListResponse(
    val stats: List<UsageStat>,
    val total: Int,
    val limit: Int,
    val offset: Int,
)

@Serializable
data class UsageSummaryItem(
    val feature: String,
    @SerialName("total_uses") val totalUses: Int,
    @SerialName("successful_uses") val successfulUses: Int,
    @SerialName("avg_duration") val avgDuration: Double,
)

@Serializable
data class UsageSummaryResponse(
    val summary: List<UsageSummaryItem>,
)

File: EyeGuideiOS/EyeGuide/Models/UsageStat.swift
import Foundation

struct UsageStat: Codable, Identifiable, Sendable {
    let id: String
    let userId: String
    let feature: String
    let timestamp: String
    let duration: Int?
    let success: Bool
    let errorMessage: String?

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case feature
        case timestamp
        case duration
        case success
        case errorMessage = "error_message"
    }
}

struct UsageStatResponse: Codable, Sendable {
    let stat: UsageStat
}

struct UsageListResponse: Codable, Sendable {
    let stats: [UsageStat]
    let total: Int
    let limit: Int
    let offset: Int
}

struct UsageSummaryItem: Codable, Sendable {
    let feature: String
    let totalUses: String
    let successfulUses: String
    let avgDuration: String

    enum CodingKeys: String, CodingKey {
        case feature
        case totalUses = "total_uses"
        case successfulUses = "successful_uses"
        case avgDuration = "avg_duration"
    }
}

struct UsageSummaryResponse: Codable, Sendable {
    let summary: [UsageSummaryItem]
}
